src/main/groovy/de/dkfz/eilslabs/batcheuphoria/execution/RestExecutionService.groovy

```diff
@@ -0,0 +1,207 @@
+package de.dkfz.eilslabs.batcheuphoria.execution
+
+import de.dkfz.eilslabs.batcheuphoria.jobs.Command
+import de.dkfz.eilslabs.batcheuphoria.execution.cluster.lsf.rest.RestCommand
+import de.dkfz.eilslabs.batcheuphoria.execution.cluster.lsf.rest.RestResult
+import de.dkfz.roddy.execution.io.ExecutionResult
+import de.dkfz.roddy.tools.LoggerWrapper
+import groovy.transform.NotYetImplemented
+import org.apache.commons.io.IOUtils
+import org.apache.http.Header
+import org.apache.http.HttpEntity
+import org.apache.http.auth.AuthenticationException
+import org.apache.http.client.methods.CloseableHttpResponse
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.config.Registry
+import org.apache.http.config.RegistryBuilder
+import org.apache.http.conn.socket.ConnectionSocketFactory
+import org.apache.http.conn.socket.PlainConnectionSocketFactory
+import org.apache.http.conn.ssl.DefaultHostnameVerifier
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.CloseableHttpClient
+import org.apache.http.impl.client.HttpClientBuilder
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager
+import org.apache.http.message.BasicHeader
+import org.apache.http.protocol.HTTP
+import org.apache.http.ssl.SSLContexts
+import org.apache.http.util.EntityUtils
+
+import javax.net.ssl.SSLContext
+import java.security.KeyStore
+
+/**
+ * Created by kaercher on 12.01.17.
+ */
+class RestExecutionService implements ExecutionService{
+
+    private static final LoggerWrapper logger = LoggerWrapper.getLogger(RestExecutionService.class.name);
+
+    public boolean isCertAuth = false
+
+    private String token = ""
+
+    private File keyStoreLocation
+    private String keyStorePassword
+
+    public final String BASE_URL
+
+    /*REST RESOURCES*/
+    public static final String RESOURCE_LOGON = "/logon"
+    public static final String RESOURCE_LOGOUT = "/logout"
+
+    RestExecutionService(String baseURL, String username, String password) throws AuthenticationException{
+        this.BASE_URL = baseURL
+        RestResult result = logon(username,password)
+        if(result.statusCode == 200){
```

Comment:
I don't know if the status code handling should not be done in logon. Throw the exception there.